In [8]:
pip install instagrapi

  Created wheel for instagrapi: filename=instagrapi-1.15.1-py3-none-any.whl size=86566 sha256=83f919d4af062aa17c5a85fe1c4787dc7228080dbc7cb3c7765c93c070c51435
  Stored in directory: c:\users\yhr63\appdata\local\pip\cache\wheels\b1\6f\3e\ffbc6a005adefe09b9f2cbd0e1ca1f2d1f7b78e6b2c7390d91
Successfully built instagrapi
  Attempting uninstall: requests
    Found existing installation: requests 2.22.0
    Uninstalling requests-2.22.0:
      Successfully uninstalled requests-2.22.0
Note: you may need to restart the kernel to use updated packages.


ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
conda 4.10.1 requires ruamel_yaml_conda>=0.11.14, which is not installed.
You should consider upgrading via the 'C:\Users\yhr63\Anaconda3\python.exe -m pip install --upgrade pip' command.


In [3]:
import pandas as pd
import  numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
from instagrapi import Client

In [56]:
data = pd.read_csv('insta_nail.csv')

In [59]:
duplicate = data[data.duplicated()]

In [52]:
data.drop_duplicates(inplace = True)

In [53]:
data.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 315 entries, 0 to 328
Data columns (total 7 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   username       315 non-null    object 
 1   location       275 non-null    object 
 2   lat            275 non-null    float64
 3   lng            275 non-null    float64
 4   comment_count  315 non-null    int64  
 5   like_count     315 non-null    int64  
 6   caption_text   315 non-null    object 
dtypes: float64(2), int64(2), object(3)
memory usage: 19.7+ KB


- 중복제거하고 315개
- 685개 추가로 붙이기

## 1. 데이터 탐색

### 결측치 파악

In [6]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000 entries, 0 to 999
Data columns (total 7 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   username       1000 non-null   object 
 1   location       960 non-null    object 
 2   lat            960 non-null    float64
 3   lng            960 non-null    float64
 4   comment_count  1000 non-null   int64  
 5   like_count     1000 non-null   int64  
 6   caption_text   1000 non-null   object 
dtypes: float64(2), int64(2), object(3)
memory usage: 54.8+ KB


location, lat, lng 40개 결측치

### 위치정보가 나오지 않은 행들의 계정정보 확인

In [7]:
data[data['location'].isnull()]['username'].value_counts()

nail_ilrang       11
nailbyheo          4
myjju_nail         3
parlerde_nail      3
dazzling_nail_     2
b_dalkom           2
cndcl98            1
love.love0604      1
namweonsun7        1
haileyy.oo         1
yeony742           1
didi_u3            1
hee6.451           1
60000nail          1
raon_happiness     1
mio_o.re           1
the9_hongdae       1
_sm_2019           1
1223thfk           1
ko.by0401          1
nailwizard_        1
Name: username, dtype: int64

nail_ilrang, nailbyheo, myjju_nail, parlerde_nail, dazzling_nail_, b_dalkom

#### 위치정보가 결측인 계정의 정보 확인

- 대부분 정상적인 네일계정 맞음 => 게시글 태그에 위치가 나와있는지 확인

In [8]:
c1.user_info_by_username('b_dalkom').dict()

NameError: name 'c1' is not defined

## 2. 텍스트 분석

### 게시글에서 해시태그만 추출

In [9]:
import re

In [10]:
for i in range(len(data)):
    data.loc[i,'hashtag'] = ' '.join(re.findall(r'#\w+',data.loc[i,'caption_text']))

### 해시태그에 '광고', '홍보' 문구가 들어가있는 경우의 인덱스 추출

In [11]:
ad_index = []

for i in range(len(data)):
    if '광고' in data.loc[i,'hashtag'] or '홍보' in data.loc[i,'hashtag'] or '대행' in data.loc[i,'hashtag']:
        ad_index.append(i)

In [12]:
ad_index

[0, 1, 2, 5, 7, 9, 10]

### 광고, 홍보 계정 제거

In [14]:
data.drop(index = ad_index,inplace = True)

In [15]:
# 제거 후 reset index
data.reset_index(drop = True, inplace = True)

### 해시태그 이름별로 숫자 카운트 => 딕셔너리에 저장

In [16]:
data['hashtag']

0      #합정네일 #상수네일 #이벤트네일 #네일아트추천 #형광네일 #미러네일 #연남동네일 ...
1      #합정네일 #상수네일 #이벤트네일 #네일아트추천 #마블네일 #미러네일 #연남동네일 ...
2      #비마이문네일 #연남동네일 #연남동네일샵 #동교동네일 #서교동네일 #홍대네일 #강남...
3      #합정네일 #상수네일 #이벤트네일 #네일아트추천 #형광네일 #미러네일 #연남동네일 ...
4      #합정네일 #상수네일 #엠보네일 #네일아트추천 #마블네일 #미러네일 #연남동네일 #...
                             ...                        
988    #달콤네일 #상수네일 #합정네일 #연남동네일 #서교동네일 #홍대ᄂ...
989    #합정네일 #상수네일 #홍대네일 #망원네일 #서교동네일 #연남네일 #신촌네일 #이대...
990    #달콤네일 #쿠앤크패디 #귀여운패디 #홍대네일샵 #덩어리...
991    #달콤네일 #상수네일 #합정네일 #연남동네일 #서교동네일 #홍대ᄂ...
992    #합정네일 #상수네일 #홍대네일 #망원네일 #서교동네일 #연남네일 #신촌네일 #이대...
Name: hashtag, Length: 993, dtype: object

In [17]:
hashtag_count = {}

In [18]:
for i in range(len(data)):
    hash_split = (data.loc[i,'hashtag']).split(' ')
    
    for k in hash_split:
        
        if k not in hashtag_count.keys():
            hashtag_count[k] = 1
        else:
            hashtag_count[k] += 1

In [19]:
hashtag_count

{'#합정네일': 656,
 '#상수네일': 657,
 '#이벤트네일': 29,
 '#네일아트추천': 14,
 '#형광네일': 13,
 '#미러네일': 12,
 '#연남동네일': 304,
 '#동교동네일': 84,
 '#서교동네일': 993,
 '#화이트네일': 20,
 '#신촌네일': 338,
 '#이대네일': 329,
 '#여의도네일': 311,
 '#마포네일': 341,
 '#홍대네일': 486,
 '#마블네일': 27,
 '#비마이문네일': 30,
 '#연남동네일샵': 32,
 '#강남네일': 2,
 '#블랙네일': 24,
 '#할로윈네일': 3,
 '#할로윈네일추천': 1,
 '#키티': 7,
 '#키티네일': 6,
 '#산리오': 6,
 '#시럽네일': 62,
 '#가을네일': 42,
 '#엠보네일': 9,
 '#서울네일': 3,
 '#8월네일': 2,
 '#여름네일': 277,
 '#내아이템들죄다알록달록': 1,
 '#다음달엔차분하게': 1,
 '#파우더네일': 15,
 '#달콤네일': 498,
 '#금칠네일': 1,
 '#홍대네일샵': 494,
 '#드로잉네일': 7,
 '#10월아트': 2,
 '#이달의아트': 80,
 '#홍대네일샵': 44,
 '#연암동네일': 1,
 '#10월이달의아트': 3,
 '#그라데이션네일': 33,
 '#시럽그라데이션': 6,
 '#포인트네일': 5,
 '#유니크네일': 19,
 '#보라네일': 6,
 '#시스루네일': 3,
 '#자개네일': 8,
 '#진주네일': 4,
 '#9월아트': 3,
 '#8월아트': 231,
 '#망원네일': 318,
 '#풀컬러네일': 12,
 '#그라네일': 3,
 '#프렌치네일아트': 6,
 '#손젤': 244,
 '#발젤': 246,
 '#꼼꼼네일': 1,
 '#당일예약가능': 297,
 '#시청네일': 4,
 '#종로네일': 3,
 '#공덕네일': 305,
 '#당산네일': 291,
 '#목동네일': 265,
 '#도트네일': 3,
 '#글리터네일':

In [103]:
len(hashtag_count)

928

#### 1번밖에 나오지 않은 해시태그는 제외

In [20]:
hashtag_count = {key: value for key, value in hashtag_count.items() if value >=2}

In [21]:
hashtag_count

{'#합정네일': 656,
 '#상수네일': 657,
 '#이벤트네일': 29,
 '#네일아트추천': 14,
 '#형광네일': 13,
 '#미러네일': 12,
 '#연남동네일': 304,
 '#동교동네일': 84,
 '#서교동네일': 993,
 '#화이트네일': 20,
 '#신촌네일': 338,
 '#이대네일': 329,
 '#여의도네일': 311,
 '#마포네일': 341,
 '#홍대네일': 486,
 '#마블네일': 27,
 '#비마이문네일': 30,
 '#연남동네일샵': 32,
 '#강남네일': 2,
 '#블랙네일': 24,
 '#할로윈네일': 3,
 '#키티': 7,
 '#키티네일': 6,
 '#산리오': 6,
 '#시럽네일': 62,
 '#가을네일': 42,
 '#엠보네일': 9,
 '#서울네일': 3,
 '#8월네일': 2,
 '#여름네일': 277,
 '#파우더네일': 15,
 '#달콤네일': 498,
 '#홍대네일샵': 494,
 '#드로잉네일': 7,
 '#10월아트': 2,
 '#이달의아트': 80,
 '#홍대네일샵': 44,
 '#10월이달의아트': 3,
 '#그라데이션네일': 33,
 '#시럽그라데이션': 6,
 '#포인트네일': 5,
 '#유니크네일': 19,
 '#보라네일': 6,
 '#시스루네일': 3,
 '#자개네일': 8,
 '#진주네일': 4,
 '#9월아트': 3,
 '#8월아트': 231,
 '#망원네일': 318,
 '#풀컬러네일': 12,
 '#그라네일': 3,
 '#프렌치네일아트': 6,
 '#손젤': 244,
 '#발젤': 246,
 '#당일예약가능': 297,
 '#시청네일': 4,
 '#종로네일': 3,
 '#공덕네일': 305,
 '#당산네일': 291,
 '#목동네일': 265,
 '#도트네일': 3,
 '#글리터네일': 26,
 '#네일': 23,
 '#패디': 8,
 '#이달의아트네일': 2,
 '#웨딩아트': 35,
 '#연남동네일샵추천': 4,
 '#빈티지네일': 3,
 '#여

In [53]:
len(hashtag_count)

401

### 위치정보 카운트해 딕셔너리로 저장

In [22]:
location_count = {}

In [23]:
for i in range(len(data)):
    location = data.loc[i,'location']
    if location not in location_count:
        location_count[location] = 1
    else:
        location_count[location] += 1

In [24]:
location_count

{nan: 34,
 '연남동': 31,
 '서교동네일': 2,
 '달콤네일': 505,
 '홍대 상수역 근처': 62,
 '홍대네일싸롱': 318,
 'Seoul, Korea': 1,
 '오유네일': 5,
 '홍대입구역 주변': 1,
 '이솝네일': 14,
 '홍대입구역': 2,
 '홍대': 3,
 '홍대베라클럽': 1,
 '합정역 주변 어딘가': 1,
 '네일에이블': 1,
 '홍대비쥬네일': 1,
 '서울 서초구': 1,
 '투더블유': 7,
 '연남동 어딘가': 2,
 '연트럴파크': 1}

In [57]:
len(location_count)

20

#### 위치에 1번밖에 나오지 않은 곳은 제외

In [25]:
location_count = {key: value for key, value in location_count.items() if value >= 2}

In [26]:
len(location_count)

12

In [27]:
location_count

{nan: 34,
 '연남동': 31,
 '서교동네일': 2,
 '달콤네일': 505,
 '홍대 상수역 근처': 62,
 '홍대네일싸롱': 318,
 '오유네일': 5,
 '이솝네일': 14,
 '홍대입구역': 2,
 '홍대': 3,
 '투더블유': 7,
 '연남동 어딘가': 2}

- 달콤네일,홍대네일싸롱, 이솝네일, 오유네일 : 실제 운영중인 가게라면 위도, 경도 정보가 있을것 => 확인
- 달콤네일, 홍대네일싸롱이 위치태그가 많이됨

### 달콤네일과 홍대네일싸롱을 위치로 태그한 사용자 조사

- 위치로 많이 태그되긴했지만, 다 네일운영자가 자신의 샵을 직접태그한것 => 핫플이라고 할 수 있는가?

In [31]:
data[data['location'] == '달콤네일']['username'].value_counts()

b_dalkom    505
Name: username, dtype: int64

In [32]:
data[data['location'] == '홍대네일싸롱']['username'].value_counts()

nail__salons    318
Name: username, dtype: int64

### 달콤네일, 홍대네일싸롱을 그룹으로 집계

In [33]:
popular_nail = data[data['location'].isin(['달콤네일','홍대네일싸롱'])]

In [34]:
popular_nail.groupby(['location','username']).mean()

lat         lng  comment_count  like_count
location username                                                      
달콤네일     b_dalkom      37.554188  126.923847       0.013861  346.380198
홍대네일싸롱   nail__salons  37.549908  126.921054       0.028302  213.710692

- 달콤네일은 업체가 직접 위치를 태그해 게시글 작성 : 평균 좋아요 346
- 홍대네일싸롱도 업체가 직접 위치를 태그해 게시글 작성 : 평균 좋아요 213

=> 좋아요수는 많지만 업체가 직접 올린 홍보글이나 마찬가지 => 이곳을 핫플이라고 할 수 있을지!!

#### 달콤네일, 홍대네일싸롱이 해시태그로 있는지 확인

In [35]:
hashtag_count['#달콤네일']

498

In [36]:
hashtag_count['#홍대네일싸롱']

8

In [40]:
hashtag_count['#네일싸롱']

7

### 해시태그 언급 횟수로 정렬해 핫플확인해보기

In [41]:
# hashtag_count 딕셔너리 정렬
hashtag_count_sorted= sorted(hashtag_count.items(), reverse=True, key=lambda item: item[1])



In [42]:
hashtag_count_sorted

[('#서교동네일', 993),
 ('#상수네일', 657),
 ('#합정네일', 656),
 ('#달콤네일', 498),
 ('#홍대네일샵', 494),
 ('#홍대네일', 486),
 ('#마포네일', 341),
 ('#신촌네일', 338),
 ('#이대네일', 329),
 ('#망원네일', 318),
 ('#여의도네일', 311),
 ('#공덕네일', 305),
 ('#연남동네일', 304),
 ('#당일예약가능', 297),
 ('#당산네일', 291),
 ('#연남네일', 280),
 ('#여름네일', 277),
 ('#목동네일', 265),
 ('#스톤네일', 252),
 ('#발젤', 246),
 ('#손젤', 244),
 ('#영등포네일', 243),
 ('#7월네일', 242),
 ('#7월아트', 238),
 ('#파츠네일', 237),
 ('#마블아트', 233),
 ('#대리석아트', 232),
 ('#파스텔네일', 232),
 ('#8월이달의아트', 232),
 ('#8월아트', 231),
 ('#여름네일아트', 231),
 ('#덩어리패디', 230),
 ('#쿠앤크패디', 229),
 ('#귀여운패디', 229),
 ('#시럽패디', 229),
 ('#8월이다릐네일', 228),
 ('#젤네일', 85),
 ('#동교동네일', 84),
 ('#이달의아트', 80),
 ('#시럽네일', 62),
 ('#네일아트', 59),
 ('#홍대네일샵', 44),
 ('#nailart', 43),
 ('#가을네일', 42),
 ('#젤패디', 41),
 ('#웨딩아트', 35),
 ('#그라데이션네일', 33),
 ('#웨딩네일', 33),
 ('#연남동네일샵', 32),
 ('#비마이문네일', 30),
 ('#이벤트네일', 29),
 ('#이달의네일', 29),
 ('#마블네일', 27),
 ('#패디아트', 27),
 ('#글리터네일', 26),
 ('#귀여운네일', 26),
 ('#이달아트', 26),
 ('#블랙

- 대부분 @@@동네일 태그가 많음
- 상호 이름으로 추정되는 곳 : 달콤네일(언급498회)비마이문네일(언급30회), 이솝네일(언급14회)

In [45]:
# 비마이문네일, 이솝네일 살펴보기

dalcom = data[data['hashtag'].str.contains('#달콤네일')== True]
bemymoon = data[data['hashtag'].str.contains('#비마이문네일') == True]
aesop = data[data['hashtag'].str.contains('#이솝네일') == True]

In [46]:
dalcom.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 498 entries, 6 to 991
Data columns (total 8 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   username       498 non-null    object 
 1   location       498 non-null    object 
 2   lat            498 non-null    float64
 3   lng            498 non-null    float64
 4   comment_count  498 non-null    int64  
 5   like_count     498 non-null    int64  
 6   caption_text   498 non-null    object 
 7   hashtag        498 non-null    object 
dtypes: float64(2), int64(2), object(4)
memory usage: 35.0+ KB


#### 달콤네일

In [117]:
dalcom['username'].value_counts()

b_dalkom    498
Name: username, dtype: int64

In [123]:
# 좋아요수
dalcom['like_count'].describe()

count    498.000000
mean     346.074297
std       24.355196
min      121.000000
25%      342.000000
50%      342.000000
75%      358.000000
max      416.000000
Name: like_count, dtype: float64

#### 비마이문네일

In [47]:
bemymoon['username'].value_counts()

bemymoon_nail    30
Name: username, dtype: int64

In [127]:
# 좋아요수
bemymoon['like_count'].describe()

count     30.000000
mean     193.566667
std       71.426685
min      106.000000
25%      141.250000
50%      186.500000
75%      221.000000
max      432.000000
Name: like_count, dtype: float64

#### 이솝네일

In [118]:
aesop['username'].value_counts()

aesop_nail    14
Name: username, dtype: int64

In [128]:
# 좋아요수
aesop['like_count'].describe()

count     14.00000
mean     144.50000
std        6.29713
min      138.00000
25%      142.00000
50%      142.00000
75%      144.75000
max      164.00000
Name: like_count, dtype: float64

#### 해시태그 많이 언급된 세곳 모두 자신들의 가게를 직접 태그한것
#### 좋아요 수는 많은편이지만 핫플이라고 할 수 있는지는 모르겠음

## insta_nail 정리

- 1) 해시태그에 많이 언급된 네일업체 : 달콤네일, 비마이문네일, 이솝네일
- 2) 위치정보에 많이 태그된 네일업체 : 달콤네일, 홍대네일싸롱
-                          
- 계정정보를 확인해본결과 달콤네일과 홍대네일싸롱이 팔로워수가 많기는함
- 해시태그나 위치정보에 많이 언급 및 태그된것이 업체에서 자기자신을 직접 태그한것
- 좋아요 수가 많기는 하지만 핫플이라고 할 수 있을지 모르겠음!!ㅠ
- 네일아트의 특성상 업체에서 시술을 받은 손님의 사진이나 이달의 디자인등 게시글을 올리면서 dm을통해 고객들과 연락
- 네일아트를 받은 고객이 직접 태그해서 글을 올리는 경우가 거의 없기는 함

달콤네일, 홍대네일싸롱 